specification/responses/5xx/502_BadGateway.yaml

```diff
@@ -0,0 +1,21 @@
+description: |+
+  NHS Notify is currently unable to communicate with a downstream API required to process your request.
```

Comment:
I think the references in the 502 and 504 responses should refer to "upstream" services rather than "downstream" because a downstream service depends on an upstream service, e.g. front-end depends on back-end, back-end depends on NHSApp etc so front-end is downstream of back-end  and back-end is downstream of NHSApp.

Or a better reasoned argument from W3C:
> The terms "upstream" and "downstream" are used to describe directional requirements in relation to the message flow: all messages flow from upstream to downstream.

and:
> The 504 (Gateway Timeout) status code indicates that the server, while acting as a gateway or proxy, did not receive a timely response from an upstream server it needed to access in order to complete the request.

Reply:
Downstream was chosen to reflect the language used in the examples I received from Simon, which in turn I believe was used to reflect the language used by the live documentation. If it is your intention to use the language upstream, would you like me to change the other parts of the documentation that currently use the language upstream and communicate to Simon and Terry that we'll be needing to change the responses that the API gives?

![Capture3](https://github.com/NHSDigital/communications-manager-api/assets/145978619/791ef7be-9b5a-4c64-b1ba-9c4125d65562)

Reply:
We discussed this on a call and agreed to keep it as downstream to avoid code changes to 502 and 504 responses for now